src/java.base/share/classes/java/lang/Integer.java

```diff
@@ -1767,6 +1767,102 @@ public static int reverse(int i) {
         return reverseBytes(i);
     }
 
+    /**
+     * Returns the value obtained by compressing the bits of the
+     * specified {@code int} value, {@code i}, in accordance with
+     * the specified bit mask.
+     * <p>
+     * For each one-bit value of the mask, {@code mb} say, from least
```

Comment:
A minor comments is that "For each one-bit value of the mask mb ...." might be a bit better, otherwise I think these methods and their javadoc looks good. If it comes up then these methods could include an example in the javadoc as they aren't hard once you see an example.